File: DSA-3/Bit Logic/power-of-2.cpp
/****

https://stackoverflow.com/questions/3638431/determine-if-an-int-is-a-power-of-2-or-not-in-a-single-line

https://www.geeksforgeeks.org/program-to-find-whether-a-no-is-power-of-two/


****/

/*This is a function problem.You only need to complete the function given below*/
//User function Template for C++
// Function to check power of two
bool isPowerofTwo(long long x){
    
    // Your code here  
    return x && !(x & (x-1));
    
}

File: Company Wise Practice/Oracle/power-of-2.java
/********

https://www.geeksforgeeks.org/program-to-find-whether-a-no-is-power-of-two/

********/

import java.util.Scanner;

class GFG {
	
	public static boolean isPowerOfTwo(Long n) {
		
		return n != 0 && ((n & (n-1)) == 0);
	}
	
	public static void main(String[] args) {
		
		Scanner sc = new Scanner(System.in);
		
		Integer t = sc.nextInt();
		
		while(t-- > 0) {
			
			Long n = sc.nextLong();
			
			if(isPowerOfTwo(n)) {
				System.out.println("YES");
			}else {
				System.out.println("NO");
			}
			
		}
		
		sc.close();
	}

}
